chirp/drivers/tk690.py

```diff
@@ -0,0 +1,1100 @@
+} embeddedMessage;
+
+#seekto 0x3ED0;
+struct {
+  u8 unknown10[10];
+  char soft[6];
+  u8 rid[10];
+  u8 unknown11[6];
+  u8 unknown12[11];
+  char soft_ver[5];
+} properties;
+
+#seekto 0x4090;
+struct {
+  char name[16];
+} grp_names[160];
+
+#seekto 0x4AA0;
+struct {
+  char name[16];
+} chs_names[160];
+
+"""
+
+MEM_SIZE = 0x60A0  # 24,736 bytes
+DAT_FILE_SIZE = 0x60E0
+ACK_CMD = b'\x06'
+RX_BLOCK_SIZE_L = 128
+MEM_LR = range(0x0380, 0x0400)
+RX_BLOCK_SIZE_M = 16
+MEM_MR = range(1, 11)
+RX_BLOCK_SIZE_H = 32
+MEM_HR = range(0, 0x2000, RX_BLOCK_SIZE_H)
+# define an empty block of data, as it will be used a lot in this code
+EMPTY_BLOCK = b"\xFF" * 256
+EMPTY_L = b"\xFF" * RX_BLOCK_SIZE_L
+EMPTY_H = b"\xFF" * RX_BLOCK_SIZE_H
+VALID_CHARS = chirp_common.CHARSET_UPPER_NUMERIC + "()/\\*@-+,.#_"
+SKIP_VALUES = ["S", ""]
+TONES = chirp_common.TONES
+DTCS_CODES = chirp_common.DTCS_CODES
+OPTSIG_LIST = ["None", "DTMF", "2-Tone 1", "2-Tone 2", "2-Tone 3"]
+
+BUTTON_FUNCTION_LIST = [
+            ('Aux A', 0), ('Aux B', 1), ('Aux C', 2),
+            ('Ch 1 direct', 3), ('Ch 2 direct', 4), ('Ch 3 direct', 5),
+            ('Ch 4 direct', 6), ('Ch 5 direct', 7), ('Ch down', 8),
+            ('Ch up', 9), ('Ch name', 10), ('Ch recall', 11), ('Del/Add', 12),
+            ('Dimmer', 13), ('Emergency Call', 14), ('Grp down', 15),
+            ('Grp up', 16), ('HC1 (fixed)', 17), ('HC2 (toggle)', 18),
+            ('Horn Alert', 19), ('Monitor', 22), ('Operator Sel tone', 23),
+            ('Public Address', 25), ('Scan', 26), ('Speaker int/ext', 28),
+            ('Squelch', 29), ('Talk Around', 30), ('no function', 255)]
+
+ASSIGNABLE_BUTTONS = ["grp_up", "grp_down", "monitor", "scan", "PF1", "PF2",
+                      "PF3", "PF4", "PF5", "PF6", "PF7", "PF8", "PF9"]
+FULL_HEAD_ONLY_BUTTONS = ["monitor", "scan", "PF6", "PF7", "PF8", "PF9"]
+
+
+def _close_radio(radio):
+    """Get the radio out of program mode"""
+    try:
+        radio.pipe.write(b"E")
+    except Exception:
+        LOG.debug("Failed to close radio, serial error")
+        raise errors.RadioError("Serial Connection Error while closing radio")
+
+
+def _checksum(data):
+    """the radio block checksum algorithm"""
+    cs = 0
+    for byte in data:
+        cs += byte
+    return cs % 256
+
+
+def _make_framel(cmd, addr):
+    """Pack the info in the format it likes"""
+    # x52 x0F (x0380-x0400)
+    return struct.pack(">BBH", ord(cmd), 0x0F, addr)
+
+
+def _make_framem(cmd, addr):
+    """Pack the info in the format it likes"""
+    # x54 x0F (x00-x0A)
+    return struct.pack(">BBB", ord(cmd), 0x0F, addr)
+
+
+def _make_frameh(cmd, addr):
+    """Pack the info in the format it likes"""
+    # x53 x8F (x0000-x2000) x20
+    return struct.pack(">BBHB", ord(cmd), 0x8F, addr, RX_BLOCK_SIZE_H)
+
+
+def _handshake(radio, msg="", full=True):
+    """Make a full handshake"""
+    if full is True:
+        # send ACK
+        radio.pipe.write(ACK_CMD)
+
+    # receive ACK
```

Comment:
I think you said you were going to work on more settings after this. When you do, would you mind removing all these pointless comments? Any comment that is meaningful (i.e. explaining why instead of what) should stay, but these "receive ACK", "check ACK", "send data" etc comments are just annoying.